packages/block-library/src/embed/variations.js

```diff
@@ -0,0 +1,377 @@
+		attributes: {
+			providerNameSlug: 'facebook',
+			previewable: false,
+			responsive: true,
+		},
+	},
+	{
+		name: 'instagram',
+		title: 'Instagram',
+		icon: embedInstagramIcon,
+		keywords: [ __( 'image' ), __( 'social' ) ],
+		description: __( 'Embed an Instagram post.' ),
+		patterns: [ /^https?:\/\/(www\.)?instagr(\.am|am\.com)\/.+/i ],
+		attributes: { providerNameSlug: 'instagram', responsive: true },
+	},
+	{
+		name: 'wordpress',
+		title: 'WordPress',
+		icon: embedWordPressIcon,
+		keywords: [ __( 'post' ), __( 'blog' ) ],
+		description: __( 'Embed a WordPress post.' ),
+		attributes: {
+			providerNameSlug: 'wordpress',
+		},
+	},
+	{
+		name: 'soundcloud',
+		title: 'SoundCloud',
+		icon: embedAudioIcon,
+		keywords: [ __( 'music' ), __( 'audio' ) ],
+		description: __( 'Embed SoundCloud content.' ),
+		patterns: [ /^https?:\/\/(www\.)?soundcloud\.com\/.+/i ],
+		attributes: { providerNameSlug: 'soundcloud', responsive: true },
+	},
+	{
+		name: 'spotify',
+		title: 'Spotify',
+		icon: embedSpotifyIcon,
+		keywords: [ __( 'music' ), __( 'audio' ) ],
+		description: __( 'Embed Spotify content.' ),
+		patterns: [ /^https?:\/\/(open|play)\.spotify\.com\/.+/i ],
+		attributes: { providerNameSlug: 'spotify', responsive: true },
+	},
+	{
+		name: 'flickr',
+		title: 'Flickr',
+		icon: embedFlickrIcon,
+		keywords: [ __( 'image' ) ],
+		description: __( 'Embed Flickr content.' ),
+		patterns: [
+			/^https?:\/\/(www\.)?flickr\.com\/.+/i,
+			/^https?:\/\/flic\.kr\/.+/i,
+		],
+		attributes: { providerNameSlug: 'flickr', responsive: true },
+	},
+	{
+		name: 'vimeo',
+		title: 'Vimeo',
+		icon: embedVimeoIcon,
+		keywords: [ __( 'video' ) ],
+		description: __( 'Embed a Vimeo video.' ),
+		patterns: [ /^https?:\/\/(www\.)?vimeo\.com\/.+/i ],
+		attributes: { providerNameSlug: 'vimeo', responsive: true },
+	},
+	{
+		name: 'animoto',
+		title: 'Animoto',
+		icon: embedAnimotoIcon,
+		description: __( 'Embed an Animoto video.' ),
+		patterns: [ /^https?:\/\/(www\.)?(animoto|video214)\.com\/.+/i ],
+		attributes: { providerNameSlug: 'animoto', responsive: true },
+	},
+	{
+		name: 'cloudup',
+		title: 'Cloudup',
+		icon: embedContentIcon,
+		description: __( 'Embed Cloudup content.' ),
+		patterns: [ /^https?:\/\/cloudup\.com\/.+/i ],
+		attributes: { providerNameSlug: 'cloudup', responsive: true },
+	},
+	{
+		// Deprecated since CollegeHumor content is now powered by YouTube
+		name: 'collegehumor',
+		title: 'CollegeHumor',
+		icon: embedVideoIcon,
+		description: __( 'Embed CollegeHumor content.' ),
+		scope: [ 'block' ],
+		patterns: [],
+		attributes: { providerNameSlug: 'collegehumor', responsive: true },
+	},
+	{
+		name: 'crowdsignal',
+		title: 'Crowdsignal',
+		icon: embedContentIcon,
+		keywords: [ 'polldaddy', __( 'survey' ) ],
+		transform: [
+			{
+				type: 'block',
+				blocks: [ 'polldaddy' ],
+				transform: ( content ) =>
+					createBlock( 'crowdsignal', {
+						content,
+					} ),
+			},
+		],
```

Comment:
To my knowledge, there are no plans for the Variations API to automatically handle per-variation transforms. And, for now, I think that's a good thing.

For the sake of avoiding complexity, maybe we could take these transforms (which are basically just renamings: `polldaddy -> crowdsignal`, `speaker -> speaker-deck`) and handle this directly in the same place (in the parser?) where we will map `core/embed-foo -> core/embed`.

Reply:
I haven't looked into more depth with this issue ( variation transforms ), but your idea definitely sounds goods.